base.py

```diff
@@ -180,14 +180,23 @@ def visit_merge_into_clause(self, merge_into_clause, **kw):
                 " SET %s" % sets if merge_into_clause.set else "")
 
     def visit_copy_into(self, copy_into, **kw):
-        formatter = copy_into.formatter._compiler_dispatch(self, **kw)
+        if hasattr(copy_into, "formatter") and copy_into.formatter is not None:
+            formatter = copy_into.formatter._compiler_dispatch(self, **kw)
+        else:
+            formatter = ""
         into = (copy_into.into if isinstance(copy_into.into, Table)
                 else copy_into.into._compiler_dispatch(self, **kw))
         from_ = None
         if isinstance(copy_into.from_, Table):
             from_ = copy_into.from_
         # this is intended to catch AWSBucket and AzureContainer
-        elif isinstance(copy_into.from_, AWSBucket) or isinstance(copy_into.from_, AzureContainer):
+        elif isinstance(
+                copy_into.from_, AWSBucket
```

Comment:
Yes, because otherwise line length would be > 190 if all conditions were on one line. I would suggest using some standard PEP-compliant code formatter like black; but didn't do this since it would introduce even more changes.